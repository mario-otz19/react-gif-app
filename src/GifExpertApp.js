import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball']; 
    const [categories, setCategories] = useState(['One Punch']);

/*     
    const handleAdd = () => {
        // categories.push('HunterXHunter');
        // setCategories(['HunterXHunter', ...categories]);
        setCategories(catgs => [ ...catgs, 'HunterXHunter']);
    }
*/

    return (
        <>
            <h2>Gif App</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />
            
            {/* <button onClick={ handleAdd }>Agregar</button> */}

            <ol>
                {
                    // El key no puede ser el índice y debe ser único
                    categories.map(category => (
                        <GifGrid 
                            key={ category }
                            category={ category } 
                        />
                    ))
                }
            </ol>
        </>
    );
}







