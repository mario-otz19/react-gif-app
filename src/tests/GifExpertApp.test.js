import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en el componente <GifExpertApp />', () => {
    
    test('Debe mostrarse correctamente', () => {
        const wrapper = shallow( <GifExpertApp /> ); 

        expect( wrapper ).toMatchSnapshot();
    });
    
    test('Debe de mostrar una lista de categorÃ­as', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> ); 

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });  
});
