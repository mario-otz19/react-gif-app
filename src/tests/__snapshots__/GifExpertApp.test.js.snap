// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    Gif App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    Gif App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
